src/odemis/driver/test/actuator_test.py

```diff
@@ -548,6 +546,67 @@ def test_metadata(self):
 
         testing.assert_pos_almost_equal(stage.speed.value, {"x": 10e-6, "y": 20e-6})
 
+    def test_update_axes_range(self):
+        """
+        Test the range calculation in ConvertStage
+        """
+        [a_min, a_max] = [0.e-3, 100.e-3]
+        [b_min, b_max] = [0.e-3, 13.e-3]
+        dependency = simulated.Stage("stage_", "test", axes=["a", "b"],
+                                     ranges={"a": [a_min, a_max], "b": [b_min, b_max]})
+        # Map dependency onto a given rotation
+        tilt = math.radians(26)
+        stage = ConvertStage("conv", "align", {"orig": dependency},
+                             axes=["a", "b"], rotation=tilt)
+        # get the range of stage
+        x_min, x_max = stage.axes["x"].range
+        y_min, y_max = stage.axes["y"].range
+
+        # Testing stage movement based on its range
+        # Move stage inside its range but outside the range of dependency
+        # Test with one of the four corner points of the rotated range of dependency
+        x_pos = 0.08987940462991671
+        y_pos = -0.04383711467890774
+        self.assertTrue(x_min <= x_pos <= x_max)
+        self.assertTrue(y_min <= y_pos <= y_max)
+        self.assertFalse(a_min <= y_pos <= a_max)
+        self.assertFalse(b_min <= y_pos <= b_max)
+        f = stage.moveAbs({"x": x_pos, "y": y_pos})
+        f.result()
+        # Move stage outside its range
+        with self.assertRaises(ValueError):
+            f = stage.moveAbs({"x": -2 * x_min, "y": 2 * y_max})
+            f.result()
+        # Move stage within its range
+        x_pos = 2e-03
+        y_pos = 0
+        self.assertTrue(x_min <= x_pos <= x_max)
+        self.assertTrue(y_min <= y_pos <= y_max)
+        f = stage.moveAbs({"x": x_pos, "y": y_pos})
+        f.result()
+        # Check if the stage is at desired location
+        testing.assert_pos_almost_equal(stage.position.value, {"x": x_pos, "y": y_pos})
+        new_x_pos = x_pos * math.cos(tilt)
+        new_y_pos = x_pos * math.sin(tilt)
+        # Check the value of dependency taking rotation into account
+        testing.assert_pos_almost_equal(dependency.position.value, {"a": new_x_pos, "b": new_y_pos})
+
+        # Testing dependency movement based on its range
+        # Move dependency on the extremes of its range
+        [a_pos, b_pos] = [a_max, b_max]
+        f = dependency.moveAbs({"a": a_pos, "b": b_pos})
+        f.result()
+        # Move outside the dep's range
+        with self.assertRaises(ValueError):
+            f = dependency.moveAbs({"a": a_max, "b": 2 * b_max})
+            f.result()
+        # Move dependency in its range
+        [a_pos, b_pos] = [100.e-3, 0.e-3]
+        self.assertTrue(a_min <= a_pos <= a_max)
+        self.assertTrue(b_min <= b_pos <= b_max)
+        f = dependency.moveAbs({"a": a_pos, "b": b_pos})
+        f.result()
```

Comment:
I find this whole testing block odd because it's testing the move only of the dependency. It should have no reason to fail. At least, that's not what you want to test in this unit test. (I guess you could argue that you want to check that the ConvertStage code didn't mess up with the dependency component, as actually it used to do, but I think that just checking the range explicitly should be really enough).